In [2]:
from ucimlrepo import fetch_ucirepo
import pandas as pd

# Télécharger le dataset
dataset = fetch_ucirepo(id=544)

# Features (variables d'entrée)
X = dataset.data.features

# Target (variable cible - niveau d'obésité)
y = dataset.data.targets

# Combiner en un seul DataFrame
df = pd.concat([X, y], axis=1)

# Vérifications
print("Dimensions du dataset :", df.shape)
print("\nPremières lignes :")
print(df.head())
print("\nNoms des colonnes :")
print(df.columns.tolist())

Dimensions du dataset : (2111, 17)

Premières lignes :
   Gender   Age  Height  Weight family_history_with_overweight FAVC  FCVC  \
0  Female  21.0    1.62    64.0                            yes   no   2.0   
1  Female  21.0    1.52    56.0                            yes   no   3.0   
2    Male  23.0    1.80    77.0                            yes   no   2.0   
3    Male  27.0    1.80    87.0                             no   no   3.0   
4    Male  22.0    1.78    89.8                             no   no   2.0   

   NCP       CAEC SMOKE  CH2O  SCC  FAF  TUE        CALC  \
0  3.0  Sometimes    no   2.0   no  0.0  1.0          no   
1  3.0  Sometimes   yes   3.0  yes  3.0  0.0   Sometimes   
2  3.0  Sometimes    no   2.0   no  2.0  1.0  Frequently   
3  3.0  Sometimes    no   2.0   no  2.0  0.0  Frequently   
4  1.0  Sometimes    no   2.0   no  0.0  0.0   Sometimes   

                  MTRANS           NObeyesdad  
0  Public_Transportation        Normal_Weight  
1  Public_Transportation 

## 📊 Résultats du chargement des données

- **Dimensions** : 2111 lignes × 17 colonnes
  - 2111 patients
  - 16 features (variables d'entrée)
  - 1 target : `NObeyesdad` (niveau d'obésité)

- **Colonnes** :
  - Variables numériques : `Age`, `Height`, `Weight`, `FCVC`, `NCP`, `CH2O`, `FAF`, `TUE`
  - Variables catégorielles : `Gender`, `family_history_with_overweight`, `FAVC`, `CAEC`, `SMOKE`, `SCC`, `CALC`, `MTRANS`
  - Variable cible : `NObeyesdad`

- **Premières observations** :
  - Les patients sont majoritairement jeunes (21-27 ans)
  - On observe différents niveaux d'obésité dès les premières lignes
  - Les données semblent propres et bien structurées ✅
  